src/AtelierTomato.Markov.Core/AuthorGroupManager.cs

```diff
@@ -61,30 +64,53 @@ public async Task AddOrUpdateAuthor(AuthorOID sender, AuthorGroupPermission auth
+
 			// All guards passed, allow write.
 			await authorGroupPermissionAccess.WriteAuthorGroupPermission(authorGroupPermission);
 		}
 
-		public async Task RemoveAuthor(AuthorOID sender, Guid groupID, AuthorOID user)
+		public async Task AcceptInvitation(AuthorOID sender, Guid ID)
+		{
+			var senderAuthorGroupPermission = await authorGroupPermissionAccess.ReadAuthorGroupPermission(ID, sender)
+			 ?? throw new ArgumentException($"Author \"{sender}\" has not been sent an invitation to group with ID \"{ID}\".", nameof(ID));
+			if (senderAuthorGroupPermission.Permissions.HasFlag(AuthorGroupPermissionType.Accepted))
+				throw new ArgumentException($"Author \"{sender}\" has already accepted invitation into group with ID \"{ID}\".", nameof(ID));
+
+			// All guards passed, allow accept.
+			senderAuthorGroupPermission.Permissions |= AuthorGroupPermissionType.Accepted;
+			await authorGroupPermissionAccess.WriteAuthorGroupPermission(senderAuthorGroupPermission);
+		}
+
+		public async Task RemoveAuthor(AuthorOID sender, Guid ID, AuthorOID user)
 		{
 			if (sender == user)
-				throw new ArgumentException($"You cannot remove yourself from a group. Please use the \"{nameof(LeaveGroup)}\" function instead.", nameof(user));
-			var senderAuthorGroupPermission = await authorGroupPermissionAccess.ReadAuthorGroupPermission(groupID, sender)
-			 ?? throw new ArgumentException($"Author \"{sender}\" is not registered to group with ID \"{groupID}\".", nameof(sender));
-			if (!senderAuthorGroupPermission.Permissions.Contains(AuthorGroupPermissionType.RemoveAuthor))
-				throw new ArgumentException($"Author \"{sender}\" does not have permission to remove authors from group with ID \"{groupID}\".", nameof(sender));
+				throw new ArgumentException($"You cannot remove yourself from a group. Please use the \"{nameof(LeaveGroupOrDenyInvitation)}\" function instead.", nameof(user));
+			var senderAuthorGroupPermission = await authorGroupPermissionAccess.ReadAuthorGroupPermission(ID, sender)
+			 ?? throw new ArgumentException($"Author \"{sender}\" is not registered to group with ID \"{ID}\".", nameof(sender));
+			if (!senderAuthorGroupPermission.Permissions.HasFlag(AuthorGroupPermissionType.Accepted))
+				throw new ArgumentException($"Author \"{sender}\" has not accepted the invitation into group with ID \"{ID}\".", nameof(sender));
+			if (!senderAuthorGroupPermission.Permissions.HasFlag(AuthorGroupPermissionType.RemoveAuthor))
+				throw new ArgumentException($"Author \"{sender}\" does not have permission to remove authors from group with ID \"{ID}\".", nameof(sender));
 			// All guards passed, allow remove.
-			await authorGroupPermissionAccess.DeleteAuthorFromAuthorGroup(groupID, user);
+			await authorGroupPermissionAccess.DeleteAuthorFromAuthorGroup(ID, user);
 		}
 
-		public async Task LeaveGroup(Guid groupID, AuthorOID sender)
+		public async Task LeaveGroupOrDenyInvitation(Guid groupID, AuthorOID sender)

```

Comment:
the name terrifies me. why are these two very different sounding operations combined like this? is there a better name or a separation in two methods possible?